File: sdk/python/src/oacp_sdk/dev_helpers.py
"""Dev registration helper."""

from __future__ import annotations

from typing import Any

from oacp_sdk.client import AgentClient
from oacp_sdk.defaults import DEFAULT_DEV_PUBLIC_KEY, PROTOCOL_VERSION


async def register_dev_agent(
    client: AgentClient,
    *,
    agent_id: str,
    name: str,
    capabilities: list[str],
    description: str | None = None,
    replace: bool = True,
) -> dict[str, Any]:
    """Register an agent using the shared development public key.

    Defaults to ``replace=True`` so demo scripts can be re-run idempotently.
    """
    identity: dict[str, Any] = {
        'id': agent_id,
        'name': name,
        'version': PROTOCOL_VERSION,
        'capabilities': capabilities,
        'publicKey': DEFAULT_DEV_PUBLIC_KEY,
    }
    if description is not None:
        identity['description'] = description
    return await client.register_agent(identity, replace=replace)

File: sdk/typescript/src/client/dev-helpers.ts
import { PROTOCOL_VERSION, type AgentIdentity } from '@oacp/core';

import { DEFAULT_DEV_PUBLIC_KEY } from '../defaults.js';
import type { AgentClient } from './agent-client.js';
import type { RegisterAgentOptions } from './types.js';

export interface RegisterDevAgentParams {
  readonly id: string;
  readonly name: string;
  readonly capabilities: readonly string[];
  readonly description?: string;
  readonly metadata?: Readonly<Record<string, unknown>>;
}

/**
 * Register an agent on a remote node using the development public key.
 * For local demos and tests only — replace with real keys in production.
 *
 * Defaults to `replace: true` so scripts can be re-run against a live server.
 */
export async function registerDevAgent(
  client: AgentClient,
  params: RegisterDevAgentParams,
  options: RegisterAgentOptions = {},
): Promise<AgentIdentity> {
  return client.registerAgent(
    {
      id: params.id,
      name: params.name,
      version: PROTOCOL_VERSION,
      capabilities: [...params.capabilities],
      publicKey: DEFAULT_DEV_PUBLIC_KEY,
      ...(params.description !== undefined ? { description: params.description } : {}),
      ...(params.metadata !== undefined ? { metadata: params.metadata } : {}),
    },
    { replace: true, ...options },
  );
}
